share/wake/lib/system/job.wake

```diff
@@ -15,41 +15,122 @@
 
 package wake
 
+export tuple JobKey =
+    export dir: String
+    export stdin: String
+    export env: String
+    export cmd: String
+    export signature: Integer
+    export visible: String
+    export isatty: Integer
+
+# Create/reserve a job handle, parameters aren't necessarily finalized
+export def primJobCreate (label: String) (jobKey: JobKey) (keep: Integer) (echo: String) (stdout: String) (stderr: String): Job =
+    def JobKey dir stdin env cmd signature visible isatty = jobKey
+
+    (\_ \_ \_ \_ \_ \_ \_ \_ \_ \_ \_ \_ prim "job_create")
+    label
+    dir
+    stdin
+    env
+    cmd
+    signature
+    visible
+    keep
+    echo
+    stdout
+    stderr
+    isatty
+
+# Imediatly complete a job with the provided ouputs without launching a process
+export def primJobVirtual (job: Job) (stdout: String) (stderr: String) (usage: Usage): Unit =
+    def Usage status runtime cputime membytes ibytes obytes = usage
+
+    (\_ \_ \_ \_ \_ \_ \_ \_ \_ prim "job_virtual")
+    job
+    stdout
+    stderr
+    status
+    runtime
+    cputime
+    membytes
+    ibytes
+    obytes
+
+# Launch the job via a child process. Values such as command or environment can be freely changed from the initial reservation.
+export def primJobLaunch (job: Job) (jobKey: JobKey) (usage: Usage): Unit =
+    def JobKey dir stdin env cmd _signature _visible isatty = jobKey
+    def Usage status runtime cputime membytes ibytes obytes = usage
+
+    (\_ \_ \_ \_ \_ \_ \_ \_ \_ \_ \_ \_ prim "job_launch")
+    job
+    dir
+    stdin
+    env
+    cmd
+    status
+    runtime
+    cputime
+    membytes
+    ibytes
+    obytes
+    isatty
+
+# Complete a job before launch with userland defined failure
+export def primJobFailLaunch (job: Job) (error: Error): Unit =
+    (\_ \_ prim "job_fail_launch") job error
+
+# Complete a job after launch with userland defined failure
+export def primJobFailFinish (job: Job) (error: Error): Unit =
+    (\_ \_ prim "job_fail_finish") job error
+
+# Complete a job successfully by providing to the runtime the inputs/outputs/usage of the job
+export def primJobFinish (job: Job) (inputs: String) (outputs: String) (all_outputs: String) (usage: Usage): Unit =
+    def Usage status runtime cputime membytes ibytes obytes = usage
+
+    (\_ \_ \_ \_ \_ \_ \_ \_ \_ \_ prim "job_finish")
+    job
+    inputs
+    outputs
+    all_outputs
```

Comment:
This is actually something I was trying to look into/figure out for the next PR because once we apply the filter at the runner level `outputs` and `all_outputs` will always be the same